### PR TITLE
RPC Refactor (5)

### PR DESCRIPTION
**Pull request by johnstonematt**
Drying up the RPC package

### CODE REVIEW

File: pkg/rpc/client.go [hunk 1]
```diff
@@ -128,3 +124,92 @@ func (c *Client) rpcRequest(ctx context.Context, data io.Reader) ([]byte, error)
...
 }
+
+func (c *Client) getResponse(ctx context.Context, method string, params []interface{}, result HasRPCError) error {
+	body, err := c.rpcRequest(ctx, formatRPCRequest(method, params))
+	// check if there was an error making the request:
+	if err != nil {
+		return fmt.Errorf("%s RPC call failed: %w", method, err)
+	}
+	// log response:
+	klog.V(2).Infof("%s response: %v", method, string(body))
```
Comment (SEJeff) [inline pkg/rpc/client.go:135]: Can this be debug please? This could be a large volume potentially.

File: pkg/rpc/client.go
Comment (SEJeff) [inline pkg/rpc/client.go:1]: Might be nice to add a comment to each of the methods with a link to the REST docs for the solana jsonrpc api.

File: pkg/rpc/client.go [hunk 2]
```diff
@@ -128,3 +124,92 @@ func (c *Client) rpcRequest(ctx context.Context, data io.Reader) ([]byte, error)
...
+		return 0, err
+	}
+	return resp.Result, nil
+}
+
+func (c *Client) GetBlockProduction(ctx context.Context, firstSlot *int64, lastSlot *int64) (*BlockProduction, error) {
+	// format params:
+	params := make([]interface{}, 1)
+	if firstSlot != nil {
+		params[0] = map[string]interface{}{
```
Comment (SEJeff) [inline pkg/rpc/client.go:187]: Why interface? Just to make it easier? The [docs](https://solana.com/docs/rpc/http/getblockproduction#result) don't look too bad.

Http json apis are 100x easier in python :)

**Review by SEJeff — APPROVED** [review]

File: pkg/rpc/client.go (hunk 2 above)
Comment (johnstonematt) [reply pkg/rpc/client.go:187]: Not sure, just used what was already in the code - will definitely like to clean up further

File: pkg/rpc/client.go (hunk 1 above)
Comment (johnstonematt) [reply pkg/rpc/client.go:135]: Klog doesn't seem to have a `Debugf` method, my impression was that the `V(2)` is the klog equivalent of debug lgos

Comment (SEJeff) [reply pkg/rpc/client.go:135]: It is such garbage 😢 

I'm going to file a ticket to replace it.

File: pkg/rpc/client.go (hunk 2 above)
Comment (SEJeff) [reply pkg/rpc/client.go:187]: Fair enough, we can do that at a later date.

**Pull request merged**